libs/generic-view/store/src/lib/views/reducer.ts

```diff
@@ -116,83 +114,62 @@ export const genericViewsReducer = createReducer(initialState, (builder) => {
   builder.addCase(getOverviewData.fulfilled, (state, action) => {
     state.lastResponse = action.payload
     const deviceId = action.payload.deviceId
-    if (deviceId) {
-      state.devices[deviceId].features = {
-        ...state.devices[deviceId].features,
-        "mc-overview": {
-          config: state.devices[deviceId].features?.["mc-overview"]?.config,
-          data: action.payload.overviewData,
-        },
-        ...(action.payload.aboutData
-          ? {
-              "mc-about": {
-                config: state.devices[deviceId].features?.["mc-about"]?.config,
-                data: action.payload.aboutData,
-              },
-            }
-          : {}),
-      }
+    state.devices[deviceId].features = {
+      ...state.devices[deviceId].features,
+      "mc-overview": {
+        config: state.devices[deviceId].features?.["mc-overview"]?.config,
+        data: action.payload.overviewData,
+      },
+      ...(action.payload.aboutData
+        ? {
+            "mc-about": {
+              config: state.devices[deviceId].features?.["mc-about"]?.config,
+              data: action.payload.aboutData,
+            },
+          }
+        : {}),
     }
   })
   builder.addCase(getOverviewConfig.fulfilled, (state, action) => {
     state.lastResponse = action.payload
     const deviceId = action.payload.deviceId
-    if (deviceId) {
-      state.devices[deviceId].features = {
-        ...state.devices[deviceId].features,
-        "mc-overview": {
-          config: action.payload.overviewConfig,
-          data: state.devices[deviceId].features?.["mc-overview"]?.data,
-        },
-        ...(action.payload.aboutConfig
-          ? {
-              "mc-about": {
-                config: action.payload.aboutConfig,
-                data: state.devices[deviceId].features?.["mc-about"]?.data,
-              },
-            }
-          : {}),
-      } as Features
-    }
-    if (state.activeDevice === undefined) {
-      state.activeDevice = deviceId
-    }
-  })
-  builder.addCase(activateDevice, (state, action) => {
-    const { deviceId } = action.payload
-    state.activeDevice = state.devices?.[deviceId]?.apiConfig
-      ? deviceId
-      : undefined
+    state.devices[deviceId].features = {
+      ...state.devices[deviceId].features,
+      "mc-overview": {
+        config: action.payload.overviewConfig,
+        data: state.devices[deviceId].features?.["mc-overview"]?.data,
+      },
+      ...(action.payload.aboutConfig
+        ? {
+            "mc-about": {
+              config: action.payload.aboutConfig,
+              data: state.devices[deviceId].features?.["mc-about"]?.data,
+            },
+          }
+        : {}),
+    } as Features
   })
   builder.addCase(removeDevice, (state, action) => {
     delete state.devices[action.payload.id]
   })
   builder.addCase(getOutboxData.fulfilled, (state, action) => {
-    const { deviceId, timestamp } = action.payload
-    if (state.activeDevice === deviceId) {
```

Comment:
I see that you removed setting value for lastRefresh field. The purpose of this field was to protect against spamming a connected device with outbox requests and it also checks if the response is not sent by the device previously selected in the case when you switch devices between creating a request and receiving a response. Are all these cases taken care of?